### PR TITLE
feat(forge): save non-broadcasted script runs under `dry-run` folder 

### LINKED ISSUE #3088: feat: prefix script outputs with something like `dry-run-*` if `--broadcast` flag is not passed
### Component

Forge

### Describe the feature you would like

Open to other prefix name ideas, but basically right now there's no way to distinguish broadcasted logs from dry runs when looking at JSON files in the `broadcast` file. It's useful to still generate them for local inspection / facilitating writing scripts/tests that read them in.

I also think we should update the default `.gitignore` on `forge init` to include `broadcast/dry-run-*.json` 

Another solution to this problem is a flag to enable/disable saving them, but that's more tedious / easy to mess up and overwrite your `run-latest.json` file

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by rkrasiuk**
close #3088
cc @mds1

### CODE REVIEW

**Review by mattsse — APPROVED** [review]
nice!

File: cli/assets/.gitignoreTemplate [hunk 1]
```diff
@@ -6,6 +6,7 @@ out/
 !/broadcast
 /broadcast/*
```
Comment (mds1) [inline cli/assets/.gitignoreTemplate:7]: Should this be removed from the default template now?

Comment (rkrasiuk) [reply cli/assets/.gitignoreTemplate]: resolved

**Comment by rkrasiuk** [comment]
i'll fix failing test shortly

**Comment by rkrasiuk** [comment]
fixed the test by implementing the fallback in case `--resume` is called with no prior `--broadcast` attempt

**Review by mattsse — APPROVED** [review]
lgtm

**Pull request merged**